drive/snippets/drive_v3/file_snippets/share_file.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// [START drive_share_file]
+
+/**
+ * Batch permission modification
+ * @param{string} realFileId file ID
+ * @param{string} realUser username
+ * @param{string} realDomain domain
+ * @return{obj} permission id
+ * */
+function shareFile(realFileId, realUser, realDomain) {
+  // Get credentials and build service
+  // TODO (developer) - Use appropriate auth mechanism for your app
+
+  const {GoogleAuth} = require('google-auth-library');
+  const {google} = require('googleapis');
+  const async = require('async');
+
+  const auth = new GoogleAuth({scopes: 'https://www.googleapis.com/auth/drive'});
+  const service = google.drive({version: 'v3', auth});
+
+  let id;
+  fileId = realFileId;
+  const permissions = [
+    {
+      'type': 'user',
+      'role': 'writer',
+      'emailAddress': 'user@example.com',
+    }, {
+      'type': 'domain',
+      'role': 'writer',
+      'domain': 'example.com',
+    },
+  ];
+  permissions[0].emailaddress = realUser;
+  permissions[1].domain = realDomain;
+  // Using the NPM module 'async'
+  try {
+    async.eachSeries(permissions, function(permission) {
```

Comment:
Was going to suggest using batch requests, but looks like the nodejs client doesn't support it :(